retrieval/discovery/ec2.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/log"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/util/strutil"
+)
+
+const (
+	ec2Label = model.MetaLabelPrefix + "ec2_"
+)
+
+// EC2Discovery periodically performs EC2-SD requests. It implements
+// the TargetProvider interface.
+type EC2Discovery struct {
+	aws            *aws.Config
+	done           chan struct{}
+	interval       time.Duration
+	port           int
+	instanceFilter []*ec2.Filter
```

Comment:
Okay, I sense that those filters only operate on tags (actually labels) too, which would be logical duplication of our relabeling, so this is safe to be dropped at the cost of initial convenience.

In that case let's drop the field here and make it clear how filtering is supposed to be done for us via a brief comment in the code of `refresh()` and by mention and example in the documentation.